### ISSUE TITLE
jail root nullfs won't unmount (device busy)

### ISSUE THREAD

**Issue by joachimmathes** [iocage/iocage#936]
- [x] Supply `iocage --version`
  Version 1.2 RC (`6474acdf65169e4cd6711d7144b9fe574d1c8397`)
- [x] Supply the commands used, along with any steps to recreate it.
  `iocage stop <JAIL>`
- [x] Provide the output from the command you issued.
  ```
  * Stopping <JAIL>
    + Successfully removed RCTL rules for <JAIL>
    + Executing prestop OK
    + Stopping services OK
    + Tearing down VNET OK
    + Removing devfs_ruleset: 8 OK
    + Removing jail process FAILED:
  umount: unmount of /iocage/jails/<JAIL>/root failed: Device busy
  jail: ioc-<JAIL>: /sbin/umount -t nullfs /iocage/jails/<JAIL>/root: failed
  ```
- [x] Supply what you expected the result or output to be
  ```
  * Stopping <JAIL>
    + Successfully removed RCTL rules for <JAIL>
    + Executing prestop OK
    + Stopping services OK
    + Tearing down VNET OK
    + Removing devfs_ruleset: 8 OK
    + Removing jail process OK
    + Executing poststop OK
  ```
- [x] Checked that the problem has not already been fixed on `master` if using
  a stable release.

The jail fstab:
```
zroot/iocage/jails/<JAIL> on /iocage/jails/<JAIL> (zfs, local, nfsv4acls)
/iocage/jails/<JAIL_TEMPLATE>/root on /iocage/jails/<JAIL>/root (nullfs, local, read-only)
zroot/iocage/jails/<JAIL>/root/etc on /iocage/jails/<JAIL>/root/etc (zfs, local, nfsv4acls)
...
```

The problem of unmounting jail directories seems to be an old and odd
one:
* https://forums.freebsd.org/threads/jail-root-nullfs-wont-unmount-device-busy.45806/
* https://forums.freebsd.org/threads/jail-mounted-base-nullfs-sometimes-not-un-mounted-after-jail-stop.69415/

And it is obviously not a problem of `iocage` itself but of the
underlying `jail` command.

Nevertheless, the problem occurred with the introduction of using
`/var/run/jail.ioc-{self.uuid}.conf` files in iocage, i.e. official
release `1.1` https://github.com/iocage/iocage/blob/3e0c61ada9a22e273590e5c579e65928c61c59b3/iocage_lib/ioc_stop.py#L309

With this configuration file `jail` takes care of unmounting the
jail directories by itself on `iocage stop`. In iocage 1.0 `jail`
took only care of killing the processes (afaik). The umounting of
directories was performed by `iocage` explicitly
https://github.com/iocage/iocage/blob/3e0c61ada9a22e273590e5c579e65928c61c59b3/iocage_lib/ioc_stop.py#L375
```
umount -afF {self.path}/fstab
```
That is why we haven't run into the problem so far.

Since the explicit umount commands are currently still
available
https://github.com/iocage/iocage/blob/6474acdf65169e4cd6711d7144b9fe574d1c8397/iocage_lib/ioc_stop.py#L393
it helps to remove the lines from `ioc_stop.py` where `-f ...`
is appended to the jail command
https://github.com/iocage/iocage/blob/6474acdf65169e4cd6711d7144b9fe574d1c8397/iocage_lib/ioc_stop.py#L32.

Long story short, the umount problem seems to be hard one within
`jail` or a deeper abstraction layer. Do you see any chance for a
workaround in `iocage`? For example, an option `-f` for `iocage stop`
to make it use the `/var/run/jail.ioc-{self.uuid}.conf` file. And no
`.conf` usage as default?

**Comment by skarekrow** [comment]
Hmm I would assume `jail` is also doing this, but we could perhaps reintroduce the umount commands in iocage as a safety net for this sort of thing. @sonicaj thoughts?

**Comment by yerrysherry** [comment]
I checked out the latest version from github and everything works fine. It is importent to use 
iocage fstab -a <jail> ... or iocage fstab -r <jail> to update the fstab. When editing with 
iocage fstab -e <jail> or using an editor (vi, emacs, ...), then it is likely that we have mount problems.

ex.

# iocage list -h
-	dns4	down	11.2-RELEASE	192.168.59.61
22	www1	up	11.2-RELEASE	192.168.59.60

# iocage fstab -e www1
#/jails/dump    /jails/iocage/jails/www1/root/dump      nullfs  rw      0       0 # Added by iocage on 2019-06-26 2
3:32:30

root@jail1:~ # iocage stop www1
* Stopping www1
  + Executing prestop OK
  + Stopping services OK
  + Removing devfs_ruleset: 10 OK
  + Removing jail process OK
  + Executing poststop OK
root@jail1:~ # mount -p -t nullfs
/jails/dump		/jails/iocage/jails/www1/root/dump nullfs	rw,noatime,nfsv4acls 	0 0
root@jail1:~ #

root@jail1:~ # iocage start www1
* Starting www1
  + Started OK
  + Using devfs_ruleset: 10
  + Using IP options: ip4.addr=em0|192.168.59.60 ip4.saddrsel=1 ip4=new ip6.saddrsel=1 ip6=new
  + Starting services OK
  + Executing poststart OK

root@jail1:~ # iocage exec  www1 df -h
Filesystem                      Size    Used   Avail Capacity  Mounted on
jails/iocage/jails/www1/root     12G    1.0G     11G     9%    /
/jails/dump                     3.0G     92K    3.0G     0%    /dump
devfs                           1.0K    1.0K      0B   100%    /dev
fdescfs                         1.0K    1.0K      0B   100%    /dev/fd

The "/jails/dump" is still active

# iocage stop  www1
* Stopping www1
  + Executing prestop OK
  + Stopping services OK
  + Removing devfs_ruleset: 10 OK
  + Removing jail process OK
  + Executing poststop OK

# mount -p -t nullfs
/jails/dump		/jails/iocage/jails/www1/root/dump nullfs	rw,noatime,nfsv4acls 	0 0
# umount /jails/iocage/jails/www1/root/dump
# mount -p -t nullfs
#

Now, the nullfs mountpoint is disappeared. 

So, be carefully when changing the fstab when the jails is running!!
